programs/nodeos/CMakeLists.txt

```diff
@@ -1,3 +1,10 @@
+include(additionalPlugin)
+
+foreach(ADDITIONAL_PLUGIN_SOURCE_DIR ${EOSIO_ADDITIONAL_PLUGINS})
+    message(STATUS "[Additional Plugin] source dir: ${ADDITIONAL_PLUGIN_SOURCE_DIR}")
+    add_subdirectory(${ADDITIONAL_PLUGIN_SOURCE_DIR} ${CMAKE_BINARY_DIR}/additional_plugins/${ADDITIONAL_PLUGIN_SOURCE_DIR})
```

Comment:
this would appear to concatenate the whole path to the additional plugin to the directory.  Will this cause problems if the path is relative aka "../../../my_plugin" ? 

When I was investigating something similar I used
```
get_filename_component(plugin_name "${external_plugin_path}" NAME)
```
to get just the last directory name in the path (aka "my_plugin" in the example) and concatenated that to the CMAKE_BINARY_DIR.  It carries the problem of all plugins needing unique directory names.

Reply:
Ah! You right.
I see 2 solution :
1. use a random string 
2. use fixed path + incremental integer (ex plugin_1, plugin_2)

Reply:
3! With a bit of magic I could use the plugin first target name (ex build/additional_plugins/sql_db_plugin)

Targets have to be unique in cmake ... so it will handle the error cases for us.

Reply:
@wanderingbort done